pkg/ddc/thin/engine.go

```diff
@@ -116,11 +116,19 @@ func Precheck(client client.Client, key types.NamespacedName) (found bool, err e
 // CheckReferenceDatasetRuntime judge if this runtime is used for handling dataset mounting another dataset.
 func CheckReferenceDatasetRuntime(client client.Client, runtime *datav1alpha1.ThinRuntime) (bool, error) {
 	dataset, err := utils.GetDataset(client, runtime.Name, runtime.Namespace)
-	if err != nil {
+	if err != nil && utils.IgnoreNotFound(err) != nil {
+		// ignore dataset not found err and try to get mounted dataset from runtime
 		return false, err
 	}
 
-	mounted := base.GetMountedDatasetNamespacedName(dataset)
+	var mounted []types.NamespacedName
+	if dataset != nil {
+		// getMountedDataset from dataset first
+		mounted = base.GetPhysicalDatasetFromMounts(dataset.Spec.Mounts)
```

Comment:
I suggest adding  more logging info for debugging.

Reply:
Done.